functions/Gemfile

```diff
@@ -14,7 +14,7 @@
 
 source "https://rubygems.org"
 
-gem "functions_framework", "~> 1.0"
+gem "functions_framework", "~> 1.4.1"
```

Comment:
This will pin to 1.4.x and prevent upgrading to 1.5. If you want to ensure we get at least 1.4.1 while preventing a semver-major update, please use:

```
gem "functions_framework", "~> 1.4", ">= 1.4.1"
```

Same for the other Gemfiles below.

Reply:
You need both clauses: `"~> 1.4", ">= 1.4.1"`